src/helpers/extrapolateFromXdr.js

```diff
@@ -9,195 +9,219 @@
 // - string: string values that appear as just plain text
 // - object: typed values always with a type and value `{type: 'code', value: 'Foo();'}`
 
-import { xdr, StrKey, Keypair, Operation } from "stellar-sdk";
+import StellarSdk from "stellar-sdk";
+import SorobanSdk from "soroban-sdk";
 import isArray from "lodash/isArray";
 import isString from "lodash/isString";
 import functionsIn from "lodash/functionsIn";
 import includes from "lodash/includes";
 import without from "lodash/without";
 
-export default function extrapolateFromXdr(input, type) {
+export default function extrapolateFromXdr(input, type, isSoroban = false) {
   // TODO: Check to see if type exists
   // TODO: input validation
 
-  let xdrObject;
-  try {
-    xdrObject = xdr[type].fromXDR(input, "base64");
-  } catch (error) {
-    throw new Error("Input XDR could not be parsed");
-  }
-
-  let tree = [{}];
-  buildTreeFromObject(xdrObject, tree[0], type);
-  return tree;
-}
-
-function buildTreeFromObject(object, anchor, name) {
-  anchor.type = name;
-
-  if (isArray(object)) {
-    parseArray(anchor, object);
-  } else if (!hasChildren(object)) {
-    anchor.value = getValue(object, name);
-  } else if (object.switch) {
-    parseArm(anchor, object);
+  let xdr, StrKey, Keypair, Operation;
+  if (isSoroban) {
+    xdr = SorobanSdk.xdr;
+    StrKey = SorobanSdk.StrKey;
+    Keypair = SorobanSdk.Keypair;
+    Operation = SorobanSdk.Operation;
   } else {
-    parseNormal(anchor, object);
```

Comment:
just moved all these helper methods w/in `extrapolateFromXdr`, only change was the xdr, StrKey etc var assignments